### PR TITLE
add lineage to sample endpoint

### PR DESCRIPTION
**Pull request by phoenixAja**
### Description

add lineage info to sample endpoint

#### Issue
[ch136408](https://app.clubhouse.io/genepi/story/136408)

### Test plan
changed tests to reflect new endpoint changes

### CODE REVIEW

**Review by shanaxel42 — APPROVED** [review]
niccccce!!

File: src/backend/aspen/app/views/sample.py [hunk 1]
```diff
@@ -84,6 +84,28 @@ def _format_gisaid_accession(
...
+                pathogen_genome.pangolin_last_updated
+            ),
+        }
+    else:
+        lineage = {
+            "lineage": None,
+            "probability": None,
+            "version": None,
+            "last_updated": None,
+        }
```
Comment (ttung) [inline src/backend/aspen/app/views/sample.py:104]: Can you add an issue that we need to handle this for `CalledConsensusGenomes`?

File: src/backend/aspen/app/views/sample.py [hunk 2]
```diff
@@ -84,6 +84,28 @@ def _format_gisaid_accession(
     return {"status": "no_info", "gisaid_id": None}
 
 
+def _format_lineage(sample: Sample):
```
Comment (ttung) [inline src/backend/aspen/app/views/sample.py]:
```suggestion
def _format_lineage(sample: Sample) -> dict[str, Any]:
```

File: src/backend/aspen/app/views/tests/test_samples_view.py [hunk 3]
```diff
@@ -25,7 +25,13 @@ def test_samples_view(
     group = group_factory()
     user = user_factory(group)
     sample = sample_factory(group)
-    uploaded_pathogen_genome = uploaded_pathogen_genome_factory(sample)
+    pathogen_genome = uploaded_pathogen_genome_factory(
```
Comment (ttung) [inline src/backend/aspen/app/views/tests/test_samples_view.py]: i don't agree with the rename of `uploaded_pathogen_genome` -> `pathogen_genome`.

File: src/backend/aspen/app/views/tests/test_samples_view.py [hunk 4]
```diff
@@ -25,7 +25,13 @@ def test_samples_view(
...
+        sample,
+        pangolin_lineage="B.1.590",
+        pangolin_probability=1.0,
+        pangolin_version="2021-04-23",
+        pangolin_last_updated=datetime.datetime.now(),
```
Comment (ttung) [inline src/backend/aspen/app/views/tests/test_samples_view.py]: Consider adding this to the factory so it doesn't have to be repeated so many times.

Comment (phoenixAja) [reply src/backend/aspen/app/views/tests/test_samples_view.py]: added

File: src/backend/aspen/app/views/tests/test_samples_view.py (hunk 3 above)
Comment (phoenixAja) [reply src/backend/aspen/app/views/tests/test_samples_view.py]: changed it back

File: src/backend/aspen/app/views/tests/test_samples_view.py (hunk 4 above)
Comment (phoenixAja) [reply src/backend/aspen/app/views/tests/test_samples_view.py]: done

File: src/backend/aspen/app/views/sample.py (hunk 1 above)
Comment (phoenixAja) [reply src/backend/aspen/app/views/sample.py:104]: Sure will do, in that case will we be getting the lineage info from a `CalledPathogenGenome`?

Comment (phoenixAja) [reply src/backend/aspen/app/views/sample.py:104]: does [this](https://app.clubhouse.io/genepi/story/136767/lineage-info-for-when-we-have-a-called-consensus-genome-instead-of-an-uploaded-pathogen-genome) work?

Comment (ttung) [reply src/backend/aspen/app/views/sample.py:104]: > Sure will do, in that case will we be getting the lineage info from a `CalledPathogenGenome`?

Yes.

> does [this](https://app.clubhouse.io/genepi/story/136767/lineage-info-for-when-we-have-a-called-consensus-genome-instead-of-an-uploaded-pathogen-genome) work? 

Perfect!

**Pull request merged**